src/Build/BackEnd/Components/Caching/ConfigCache.cs

```diff
@@ -188,16 +188,19 @@ public bool HasConfiguration(int configId)
         /// </summary>
         public void ClearConfigurations()
         {
-            if (_configurations != null)
+            lock (_configurations)
```

Comment:
Should this `lock` be inside the null check? I'm not clear on what happens if you `lock (null)`.